src/Microsoft.OData.Core/Batch/ODataBatchWriter.cs

```diff
@@ -499,7 +499,7 @@ public virtual Task OnInStreamErrorAsync()
         /// <param name="newState">The writer state to transition into.</param>
         protected void SetState(BatchWriterState newState)
         {
-            this.InterceptException(() => this.ValidateTransition(newState));
+            this.InterceptException((thisParam, newStateParam) => thisParam.ValidateTransition(newStateParam), newState);

```

Comment:
Since this is about allocating Lambdas, and this Lambda has no free variables, can this also be created statically

```
this.InterceptException(ValidateTransition), newState);
```

```
private static ValidateTransition(??? thisParam, ??? newStateParam) => thisParam.ValidateTransition(newStateParam);
```